cleanlab_studio/studio/trustworthy_language_model.py

```diff
@@ -46,21 +63,147 @@ def __init__(self, api_key: str, quality_preset: QualityPreset) -> None:
 
         self._quality_preset = quality_preset
 
+        self._event_loop = asyncio.get_event_loop()
+        self._query_semaphore = asyncio.Semaphore(max_concurrent_requests)
```

Comment:
do we want to limit concurrency with the asyncio semaphore? if i set it to a high value, i.e. 1000, i get the generic API error. note: it doesn't hit the RateLimitError, which seems to be the expected one based on the error message suggesting to lower `max_concurrent_requests` . not sure how likely it is that the typical user will be setting this value or just using the default

<img width="1009" alt="Screenshot 2023-12-01 at 12 08 49 PM" src="https://github.com/cleanlab/cleanlab-studio/assets/13542329/4b24eafd-80ec-44c3-a902-4ee29b81f485">